components/camel-restlet/src/main/java/org/apache/camel/component/restlet/DefaultRestletBinding.java

```diff
@@ -165,13 +170,37 @@ public void populateRestletRequestFromExchange(Request request, Exchange exchang
         // Use forms only for PUT, POST and x-www-form-urlencoded
         if ((Method.PUT == method || Method.POST == method) && MediaType.APPLICATION_WWW_FORM.equals(mediaType, true)) {
             form = new Form();
-            // must use string based for forms
-            String body = exchange.getIn().getBody(String.class);
-            if (body != null) {
-                form.add(body, null);
+            
+            if (exchange.getIn().getBody() instanceof Map) {
+                //Body is key value pairs
+                try {
+                    Map pairs = exchange.getIn().getBody(Map.class);
+                    for (Object key: pairs.keySet()) {
+                        Object value = pairs.get(key);
+                        form.add(key.toString(), value != null ? value.toString() : null);
+                    }
+                } catch (Exception ex) {
+                    LOG.error("body for " + MediaType.APPLICATION_WWW_FORM + " must be Map<String,String> or string format like name=bob&password=secRet", ex);
```

Comment:
Do not swallow exception. You can rethrow the exception with the given error message, but the request must fail with an exception so the end user can react.